students/zandra_eng/finalproject/web.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python 3
+
+"""
+Reference:
+csv file
+https://docs.python.org/2/library/csv.html
+
+matplotlib
+https://matplotlib.org/gallery/pyplots/pyplot_formatstr.html#sphx-glr-gallery-pyplots-pyplot-formatstr-py
+
+
+"""
+
+
+#import datetime
+import sys
+from dateutil.parser import parse
+import matplotlib.pyplot as plt
+import csv
+
+# if pulling data from the internet for python 2 use: import pandas.io.data as web
+
+from pandas_datareader import data as web
+
+
+class StockPlotter:
+
+    def fetch_stock(self, stock_name, start, end):
+        df = web.DataReader(stock_name, "google", start, end)
+        return df
+
+    def save(self, stocks, filename):
+        stocks.to_csv(filename)
+
+    def plot(self, stock_name, csvfile):
+        dates = []
+        values = []
+        with open(csvfile) as csvfile:
```

Comment:
Couldn't you plot the DataFrame directly? might as well re-use it if you already created it.

Then have another method for loading up a dataframe from a csv file instead of getting it from the web.